pkg/platform/http/agent.go

```
@@ -0,0 +1,769 @@
+	"go.keploy.io/server/v2/pkg/agent/hooks"
+	"go.keploy.io/server/v2/pkg/client/app"
+	"go.keploy.io/server/v2/pkg/models"
+	kdocker "go.keploy.io/server/v2/pkg/platform/docker"
+	"go.keploy.io/server/v2/utils"
+	"go.uber.org/zap"
+	"golang.org/x/sync/errgroup"
+)
+
+type AgentClient struct {
+	logger       *zap.Logger
+	dockerClient kdocker.Client //embedding the docker client to transfer the docker client methods to the core object
+	apps         sync.Map
+	client       http.Client
+	conf         *config.Config
+}
+
+//go:embed assets/initStop.sh
+var initStopScript []byte
+
+func New(logger *zap.Logger, client kdocker.Client, c *config.Config) *AgentClient {
+
+	return &AgentClient{
+		logger:       logger,
+		dockerClient: client,
+		client:       http.Client{},
+		conf:         c,
+	}
+}
+
+func (a *AgentClient) GetIncoming(ctx context.Context, id uint64, opts models.IncomingOptions) (<-chan *models.TestCase, error) {
+	requestBody := models.IncomingReq{
+		IncomingOptions: opts,
+		ClientID:        id,
+	}
+
+	requestJSON, err := json.Marshal(requestBody)
+	if err != nil {
+		utils.LogError(a.logger, err, "failed to marshal request body for incoming request")
+		return nil, fmt.Errorf("error marshaling request body for incoming request: %s", err.Error())
+	}
+
+	req, err := http.NewRequestWithContext(ctx, "POST", fmt.Sprintf("http://localhost:%d/agent/incoming", a.conf.Agent.Port), bytes.NewBuffer(requestJSON))
+	if err != nil {
+		utils.LogError(a.logger, err, "failed to create request for incoming request")
+		return nil, fmt.Errorf("error creating request for incoming request: %s", err.Error())
+	}
+	req.Header.Set("Content-Type", "application/json")
+
+	// Make the HTTP request
+	res, err := a.client.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get incoming: %s", err.Error())
+	}
+
+	// Ensure response body is closed when we're done
+	go func() {
+		<-ctx.Done()
+		if res.Body != nil {
+			_ = res.Body.Close()
```

Don't ignore the error